src/RazorSdk/SourceGenerators/RazorSourceGenerator.cs

```diff
@@ -22,11 +22,17 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
             var razorSourceGeneratorOptionsWithDiagnostics = context.AnalyzerConfigOptionsProvider
                 .Combine(context.ParseOptionsProvider)
                 .Select(ComputeRazorSourceGeneratorOptions);
-            var razorSourceGeneratorOptions = razorSourceGeneratorOptionsWithDiagnostics.ReportDiagnostics(context);
+            var liveRazorSourceGeneratorOptions = razorSourceGeneratorOptionsWithDiagnostics.ReportDiagnostics(context);
 
-            var sourceItemsWithDiagnostics = context.AdditionalTextsProvider
+            var analyzerConfigOptions = context.AnalyzerConfigOptionsProvider.AsCachedIfSuppressed(liveRazorSourceGeneratorOptions);
+            var additionalTexts = context.AdditionalTextsProvider.AsCachedIfSuppressed(liveRazorSourceGeneratorOptions);
+            var parseOptions = context.ParseOptionsProvider.AsCachedIfSuppressed(liveRazorSourceGeneratorOptions);
+            var compilation = context.CompilationProvider.AsCachedIfSuppressed(liveRazorSourceGeneratorOptions);
+            var razorSourceGeneratorOptions = liveRazorSourceGeneratorOptions.WithLambdaComparer((l, r) => l.EqualsIgnoringSupression(r), l => l.GetHashCode());
```

Comment:
>l.EqualsIgnoringSupression(r) [](http://example.com/codeflow?start=107&length=29)

Why are we ignoring suppression here rather than changing the `RazorSourceGeneratorOptions`? When do the options change?

Reply:
When we go from suppressed -> unsuppressed, we don't want to note the options as having been changed unless any of the other options actually have changed. 

Thinking about it, I wonder if it might be more obvious if we break out the suppression from the rest of the options. I'll take a quick look what that might look like

Reply:
Ok, have broken the suppression out into its own input which simplifies the options significantly